contracts/Arbitrum_SpokePool.sol

```diff
@@ -108,5 +108,5 @@ contract Arbitrum_SpokePool is SpokePool {
     }
 
     // Apply AVM-specific transformation to cross domain admin address on L1.
-    function _requireAdminSender() internal override onlyFromCrossDomainAdmin {}
+    function _requireAdminSender() internal override onlyFromCrossDomainAdmin nonReentrant {}
```

Comment:
so this is an internal method. why add it here? dont we only need it on the external/public functions?

Reply:
Ah yes i'll add a comment. Basically we don't add the `nonReentrant` modifier to `onlyAdmin` functions in the base `SpokePool` contract because the `Polygon_SpokePool` will call these methods internally via the `processMessageFromRoot`. The other spoke pools like `Optimism_SpokePool` and `Arbitrum_SpokePool` have their admin functions triggered by an external contract so we should be reentrancy guarding those methods. However, in the `Polygon_SpokePool` case we need to reentrancy guard at the `processMessageFromRoot` method instead of at the admin functions.

Reply:
But this isn't equivalent to adding a re-entrancy guard to the admin function, right? It guarantees that this function wasn't re-entered, but it doesn't help us if that admin function re-enters a different function since it doesn't wrap the entire method, no?